src/libultra_code/osAiSetNextBuffer.c

```diff
@@ -0,0 +1,25 @@
+#include <global.h>
+#include <ultra64/hardware.h>
+
+s32 osAiSetNextBuffer(void* buf, u32 size) {
+    static u8 D_80130500 = false;
+    u32 bufAdjusted = (u32)buf;
+    s32 status;
+
+    if (D_80130500) {
+        bufAdjusted = (u32)buf - 0x2000;
+    }
+    if ((((u32)buf + size) & 0x1FFF) == 0) {
+        D_80130500 = true;
+    } else {
+        D_80130500 = false;
+    }
+
+    if (status = HW_REG(AI_STATUS_REG, s32), status & AI_STATUS_AI_FULL) {
```

Comment:
Is the temp needed here? If it is I think it would be better outside the condition, if possible.